keps/0002-controller-manager.md

```diff
@@ -269,20 +267,54 @@ Hopefully the content previously contained in [umbrella issues][] will be tracke
 
 [umbrella issues]: https://github.com/kubernetes/kubernetes/issues/42752
 
-## Implementation History
+### Graduation to Beta
+
+As part of the graduation to `stable` or General Availability (GA), we have set
+both process and technical goals.
+
+#### Process Goals
+
+- 
+
+We propose the following repository structure for the cloud providers which
+currently live in `kubernetes/pkg/cloudprovider/providers/*`
+
+```
+git@github.com:kubernetes/wg-cloud-provider
+git@github.com:kubernetes/aws-cloud-provider
```

Comment:
We should also make sure we get buy in from each of the specific sigs for these repos. They should be "participating sigs" on the metadata.

Reply:
Seems like azure should be owned by sig-azure, aws by sig-aws, gce by sig-gcp, openstack by sig-openstack. Not sure who we want owning ovirt, photon and vsphere but the owner should be probably be replaced by the appropriate CP sig when/if they come into existence.

Reply:
I'm not sure I understand what additional "buy in" we're looking for or trying communicate, @jbeda . The extraction of the concrete implementations within `pkg/cloudprovider` is an effort which is being coordinated by and staffed from WG Cloud Provider under it's [current charter][] which is why currently anyway all the proposed repositories are [subprojects of WG Cloud Provider][] rather than any of the SIGs. The purpose of _this_ PR is to signal agreement on the proposed repository structure required to effect the extraction of the existing providers under `pkg/cloudprovider`. 

Personally I think it would be great to have implementations of `CloudProvider` be maintained by SIGs focused on individual clouds, but I don't believe an agreement from each SIG to maintain a `CloudProvider` implementation should block the documentation of a previously made decision about the repository structure for the existing in tree providers. SIGs as they desire are free to add an alias entry in the `OWNERS` files for each implementation (currently none have) and/or list the repos once created as their own subprojects, but today there is no SIG VMWare for instance to "own" the `vsphere`, and `photon` implementations, nor is there a SIG CloudStack to own the `cloudstack` implementation. The work described in this KEP represent the table stakes required to enable such a world and I would love to move any discussions about how SIGs express their sponsorship of the extraction effort more broadly or the ongoing maintenance of the `CloudProvider` implementations  to another forum, perhaps the [mailing list][] would be a good place if not the biweekly meetings of WG Cloud Provider. That discussion would likely inform how we expect to handle implementations which aren't in tree today.

[current charter]: https://docs.google.com/document/d/1m4Kvnh_u_9cENEE9n1ifYowQEFSgiHnbw43urGJMB64/edit#heading=h.rkwrsiw228ev

[subprojects of WG Cloud Provider]: https://github.com/kubernetes/community/tree/master/wg-cloud-provider#subprojects

[mailing list]: https://groups.google.com/forum/#!forum/kubernetes-wg-cloud-provider

Reply:
I’m sorry you are reading this as “don’t do this”.  That isn’t what I’m saying.

First, a working group is not a SIG. The big difference is that WGs can’t own code. I don’t think anyone has talked about “subprojects” of a WG.  If we want the WG to own code we should talk about it as a SIG.  I’d be 100% supportive of creating SIG-cloudprovider and moving the cloud provider SIGs under that.  But that is a larger conversation than this PR, of course.

My main point is that we’ve been talking for a long time about mapping code to SIGs.  As part of the process for creating new repositories under the `kubernetes-sigs` org we required the SIG that is asking to go through the process of claiming the subprojects and repos that that SIG owns.  While this isn’t exactly the same, the idea is that we want to make forward progress in creating clear lines of sight between code and SIGs.

If we have code that no SIG is willing to claim then we should have the discussion of deprecating/deleting that code.

Regardless, as we create these repos, we’ll have to decide what to put in the top level OWNERS files for those.  If we don’t have something to put there then we should probably back up and take a close look at supporting that code.  This isn’t a new problem but the movement of this code to new repos is highlighting it.

In any case, wrt the KEP process all of this is provisional right now until the KEP moves to `implementable`. Don’t let me stop you form committing this PR. But perhaps a TODO would be useful here?  

(I’m also surprised that the various cloud provider SIGs are not listed as participating SIGs in the metadata for the KEP but I’ll just assume that is an oversight as we figure out the process around KEPs.)

Reply:
@jbeda @cheftako @calebamiles 

This is a multi-stage process. We should definitely be looking at removing this code eventually.

I agree with @jbeda that WGs don't own code. But I also agree with @calebamiles that this needs to move forward.

@calebamiles or @cheftako Please send email to the existing CP SIGs that this is happening and that they will own the code, and cc: me. https://github.com/kubernetes/community/blob/master/sigs.yaml will also need to be updated to list their new subprojects.

Since our model current is SIG per provider, we should ask the VMWare/photon maintainers to create a SIG that could own their code, which would also create a clear mechanism for contacting the owners. @luomiao @kerneltime @abrarshivani @baludontu @divyenpatel @imkin

We should similarly ask the community whether anyone wants to own ovirt and/or cloudstack. If they aren't maintained, we should nuke them. We don't have the ability to test them, anyway.

In terms of repo naming, I would like to put cloud-provider first (e.g., cloud-provider-gce), so that they are all adjacent when sorted.

Reply:
@bgrant0607, @cheftako, @jbeda PR to move existing implementations under SIG management [is here][]

[is here]: https://github.com/kubernetes/community/pull/1862